src/mono/wasm/host/DevServer/DevServerStartup.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.WebAssembly.AppHost;
+
+namespace Microsoft.WebAssembly.AppHost.DevServer;
+
+internal sealed class DevServerStartup
+{
+    public DevServerStartup(IConfiguration configuration)
+    {
+        Configuration = configuration;
+    }
+
+    public IConfiguration Configuration { get; }
+
+    public static void ConfigureServices(IServiceCollection services)
+    {
+        services.AddRouting();
+    }
+
+    public static void Configure(IApplicationBuilder app, TaskCompletionSource<ServerURLs> realUrlsAvailableTcs, ILogger logger, IHostApplicationLifetime applicationLifetime, IConfiguration configuration)
+    {
+        app.UseDeveloperExceptionPage();
+        EnableConfiguredPathbase(app, configuration);
+
+        app.UseWebAssemblyDebugging();
+
+        bool applyCopHeaders = configuration.GetValue<bool>("ApplyCopHeaders");
+
+        if (applyCopHeaders)
+        {
+            app.Use(async (ctx, next) =>
+            {
+                if (ctx.Request.Path.StartsWithSegments("/_framework") && !ctx.Request.Path.StartsWithSegments("/_framework/blazor.server.js") && !ctx.Request.Path.StartsWithSegments("/_framework/blazor.web.js"))
```

Comment:
Do we need more files for non-blazor ?

Reply:
I don't think we have any files that we want to not apply cors on

Reply:
What's special about the two files?

Reply:
Blazor has this condition in code that was shared between blazor server and webassembly. And when I added support for COPS to devserver, I duplicated the same condition, but it is probably not needed there.